File: conformance/bundles/06-doc-string-mismatch/echo.steps.ts
import { steps } from '@varar/varar'

const { sensor } = steps(() => ({}))

// Returns the WRONG string (bare — the doc string is the only slot); the core
// compares it to the doc string and throws DocStringMismatchError → trace
// failure.kind "doc-string-mismatch".
sensor('I echo the following:', (_ctx, _doc: string) => 'goodbye')

File: conformance/bundles/06-doc-string-mismatch/echo.steps.cs
// C# sibling of echo.steps.ts / .rs (bundle 06-doc-string-mismatch).
using Varar;
using Varar.Core;

namespace Varar.Corpus.B06;

public static class EchoSteps
{
    public static void Register(Steps s)
    {
        // Returns the WRONG string (bare — the doc string is the only slot); the core compares it to
        // the doc string and throws DocStringMismatchError.
        s.Sensor("I echo the following:", (state, doc) => Value.Of("goodbye"));
    }

    public static Value State() => Value.Null;
}
